### DISCUSSION TITLE
Extending AbstractController a good practice?

### DISCUSSION THREAD

**Discussion by stephanvierkant** [symfony/symfony#48161] [Q&A] [answered]
I've got something like this:
```
namespace App\Controller;

abstract class Controller extends \Symfony\Bundle\FrameworkBundle\Controller\AbstractController
{
    // some helper methods
}
final class FrontController extends Controller
{
}
```

This used to work fine, but I'm beta testing Symfony 6.2 and it's new [PSR-4 Route loading](https://symfony.com/blog/new-in-symfony-6-2-psr-4-route-loader) and now I'm getting this message:

>   Annotations from class "App\Controller\Controller" cannot be read as it is   
>   abstract in {"path":"..\/..\/src\/Controller\/","namespace":"App\\Controlle  
>   r"} (which is being imported from "/home/runner/work/qurplus/qurplus/config  
>   /routes/annotations.php"). Make sure there is a loader supporting the "anno  
>   tation" type.    

This makes me wonder if extending the AbstractController is considered a good practice. Of course it doesn't contain any business logic, but only some helpers methods.

Is extending the AbstractController a good idea? And if not, what's the best way to organize these helper methods?

**Comment by ahundiak** [comment]
Perhaps you could list a few of your base controller helper methods?  Wanting your own base controller with commonly used methods is quite natural and common but it turns out that there are often other approaches that work better.  In some cases, controller listeners might be the way to go.  Traits can also be very handy.  Hard to tell without specifics.

As far as the error goes, I have not tried the new routing stuff but purely as a guess you might try moving your abstract base controller to a completely different directory just to see if the annotation reader is getting confused by the abstract nature of your class.

**Comment by MatTheCat** [comment, marked as answer]
Actually [the commit](https://github.com/symfony/symfony/commit/946ccb6bd287c46bccf2f4f94a0ddded33e1c0c6) which made sure you could extend your own abstract controller is the one which makes it crash today :sweat_smile: 

So yes you should be able to do it. I opened https://github.com/symfony/symfony/issues/48168

**Reply by MatTheCat** [reply]
https://github.com/symfony/symfony/pull/48170 got merged so you should be good with the next 6.2 release :ok_hand:

**Reply by stephanvierkant** [reply]
Thanks!

**Reply by MatTheCat** [reply]
Did this answer your question? If so you can mark it as accepted :ok_hand:
